src/components/Sidebar.vue

```diff
@@ -0,0 +1,84 @@
+<template>
+  <v-navigation-drawer v-model="show" temporary app>
+    <v-list-item>
+      <v-list-item-content>
+        <v-list-item-title class="title">Enspire</v-list-item-title>
+        <v-list-item-subtitle>Make CAS great again</v-list-item-subtitle>
+      </v-list-item-content>
+    </v-list-item>
+    <v-divider />
+    <v-list nav dense>
+      <v-list-item-group
+        v-model="group"
+        active-class="primary--text text--accent-4"
+      >
+        <v-list-item to="/">
+          <v-list-item-icon>
+            <v-icon>mdi-home</v-icon>
+          </v-list-item-icon>
+          <v-list-item-content>
+            <v-list-item-title>首页</v-list-item-title>
+          </v-list-item-content>
+        </v-list-item>
+        <v-divider />
+        <v-subheader>CAS管理</v-subheader>
+        <v-list-item
+          v-for="item in drawerItems"
+          :key="item.title"
+          :to="item.to"
+        >
+          <v-list-item-icon>
+            <v-icon>{{ item.icon }}</v-icon>
+          </v-list-item-icon>
+          <v-list-item-content>
+            <v-list-item-title>{{ item.title }}</v-list-item-title>
+          </v-list-item-content>
+        </v-list-item>
+        <v-divider />
+        <v-subheader>信息</v-subheader>
+        <v-list-item to="/about">
+          <v-list-item-icon>
+            <v-icon>mdi-information</v-icon>
+          </v-list-item-icon>
+          <v-list-item-content>
+            <v-list-item-title>关于我们</v-list-item-title>
+          </v-list-item-content>
+        </v-list-item>
+      </v-list-item-group>
+    </v-list>
+  </v-navigation-drawer>
+</template>
+
+<script lang="ts">
+import Vue from "vue";
+
+export default Vue.extend({
+  props: {
+    drawer: Boolean,
+  },
+  data() {
+    return {
+      group: null,
+      drawerItems: [
+        { title: "我们的社团", icon: "mdi-view-dashboard", to: "/clubs" },
+        { title: "预约教室", icon: "mdi-calendar", to: "/reservation" },
+      ],
+    };
+  },
+  computed: {
+    show: {
+      get() {
+        return (this as Vue).$props.drawer;
+      },
+      set(newValue) {
+        if (newValue === false) (this as Vue).$emit("hide-sidebar");
+      },
+    },
+  },
```

Comment:
Change them to
```typescript
computed: {
  show: {
    get(): boolean {
      return (this as Vue).$props.drawer;
    },
    set(newValue: boolean) {
      if (newValue === false) (this as Vue).$emit("hide-sidebar");
    },
  },
},
```
to resolve the `vti` errors
```
Error: Property 'show' does not exist on type 'CombinedVueInstance<Vue, object, object, object, Record<never, any>>'.
Error: Property 'group' does not exist on type 'CombinedVueInstance<Vue, object, object, object, Record<never, any>>'.
Error: Property 'drawerItems' does not exist on type 'CombinedVueInstance<Vue, object, object, object, Record<never, any>>'.
```

For details, refer to https://github.com/vuejs/vue/issues/8721#issuecomment-545671348

Reply:
Oh, okay... I read that thread but it was too long. I'm thinking about refactoring the code sometime so that it conforms to typescript conventions.